### PR TITLE
[Enhance] Company wise perpetual inventory setting

### PR DESCRIPTION
**Pull request by rohitwaghchaure**
Added perpetual inventory setting in the company
perpetual-1

### CODE REVIEW

File: erpnext/controllers/stock_controller.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def make_gl_entries(self, gl_entries=None, repost_future_gle=True, from_repost=F
 		if self.docstatus == 2:
 			delete_gl_entries(voucher_type=self.doctype, voucher_no=self.name)
 
-		if cint(frappe.defaults.get_global_default("auto_accounting_for_stock")):
+		if cint(frappe.db.get_value('Company', self.company, 'enable_perpetual_inventory')):
```
Comment (rmehta) [inline erpnext/controllers/stock_controller.py]: this function is called so many times!

It should be `erpnext.is_perpetual_inventory_enabled(company)` and the result must be cached locally for the company.

File: erpnext/stock/doctype/delivery_note/delivery_note.js [hunk 2]
```diff
@@ -225,6 +225,24 @@ erpnext.stock.DeliveryNoteController = erpnext.selling.SellingController.extend(
...
+		}
+	},
+
+	company: function(frm) {
+		frm.trigger("unhide_account_head");
+	},
+	
+	unhide_account_head: function(frm) {
+		// unhide expense_account and cost_center if perpetual inventory is enabled in the company
+		var aii_enabled = frappe.get_doc(":Company", frm.doc.company).enable_perpetual_inventory
```
Comment (rmehta) [inline erpnext/stock/doctype/delivery_note/delivery_note.js]: this should also be a library method `erpnext.is_perpetual_inventory_enabled`

File: erpnext/setup/doctype/company/company.py [hunk 3]
```diff
@@ -146,7 +146,7 @@ def set_default_accounts(self):
 		self._set_default_account("accumulated_depreciation_account", "Accumulated Depreciation")
 		self._set_default_account("depreciation_expense_account", "Depreciation")
 
-		if cint(frappe.db.get_single_value("Accounts Settings", "auto_accounting_for_stock")):
+		if self.enable_perpetual_inventory:
```
Comment (rmehta) [inline erpnext/setup/doctype/company/company.py]: any validation when I switch it on or off?

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by rmehta** [comment]
Please fix, rebase and re-open

**Comment by rohitwaghchaure** [comment]
Hi @rmehta 
Have no rights to reopen the PR, therefore created a new PR https://github.com/frappe/erpnext/pull/9355

**Comment by rmehta** [comment]
You cannot re-open because it has conflicts (I think) or you have not pushed any update.

**Pull request closed without merging**